griptape/drivers/vector/pgvector_vector_store_driver.py

```diff
@@ -148,8 +149,19 @@ def query(
             # The query should return both the vector and the distance metric score.
             query_result = session.query(self._model, op(vector).label("score")).order_by(op(vector))  # pyright: ignore
 
+            filter_kwargs: Optional[OrderedDict] = None
+
             if namespace:
-                query_result = query_result.filter_by(namespace=namespace)
+                filter_kwargs = OrderedDict(namespace=namespace)
+
+            if "filter" in kwargs and isinstance(kwargs["filter"], dict):
+                if filter_kwargs:
+                    filter_kwargs.update(kwargs["filter"])
+                else:
+                    filter_kwargs = OrderedDict(kwargs["filter"])
+
+            if filter_kwargs:
```

Comment:
Be explicit when checking for `None` https://google.github.io/styleguide/pyguide.html#2144-decision